src/Development/IDE/GHC/Compat.hs

```diff
@@ -380,6 +384,14 @@ instance HasSrcSpan (GenLocated SrcSpan a) where
 getHeaderImports a b c d =
     catch (Right <$> Hdr.getImports a b c d)
           (return . Left . srcErrorMessages)
+
+-- | Add the @-boot@ suffix to all output file paths associated with the
+-- module, not including the input file itself
+addBootSuffixLocnOut :: GHC.ModLocation -> GHC.ModLocation
+addBootSuffixLocnOut locn
+  = locn { ml_hi_file  = Module.addBootSuffix (ml_hi_file locn)
+         , ml_obj_file = Module.addBootSuffix (ml_obj_file locn)
+         }
```

Comment:
Why is the field for hie files (`ml_hie_file`) not handled in the same way?

Reply:
If you expand the fold a bit above, you will find that this bit is in the `#else` block of a `#if MIN_GHC_API_VERSION(8,8,0)`. So the answer is that `.hie` files (and the `ml_hie_file` field) don't exist in this piece of code.

See also the definition of `ml_hie_file` as a function for GHC < 8.8 above.